sotrplib/maps/core.py

```diff
@@ -290,6 +296,9 @@ def get_snr(self):
     def get_flux(self):
         with np.errstate(divide="ignore"):
             flux = self.intensity
+        ## assume that if matched filtered, intensity is rho and ivar is kappa
```

Comment:
I'm not sure this is a good way to handle things. I think that the preprocessing should potentially return a new RhoKappaMap object. So:

- Read in I, Ivar as object X
- Preprocess -> produces object Y which is a RhoKappaMap
- Continue with the rest of the pipeline.